### ISSUE TITLE
Keep selection when scrolling in copy-mode

### ISSUE THREAD

**Issue by choco** [tmux/tmux#642]
**Platform:** OS X
**Version:** tmux git afdef6d122ab667a48b843af74541e9aafcf98e6 (last commit)
**Terminal emulator:** iTerm2
**$TERM:**
- outside: `screen-256color`
- inside: `screen-256color`

I have the following configuration in my tmux.conf:
```
bind -T copy-mode-vi MouseDrag1Pane      send -X begin-selection
unbind  -T copy-mode-vi MouseDragEnd1Pane
```
This allows me to keep a selection without clearing it immediately.
When I create a selection using the bind above and then try to scroll, the selection will get expanded from where the scroll is initiated. I use the following slightly modified bind but the default also triggers the issue
```
bind -T copy-mode-vi WheelUpPane         send -X -N 3 scroll-up
bind -T copy-mode-vi WheelDownPane       send -X -N 3 scroll-down
```
Basically i'd like to be able to scroll and not mess with current selection if possible.

**Comment by wincent** [comment]
@choco: I noticed the same. I'm avoiding the expansion [like this](https://github.com/wincent/wincent/blob/7537a6db15d5b8e62560be5ffab31908191e8ac3/roles/dotfiles/files/.tmux.conf#L69-L71):

```
bind-key -T copy-mode-vi WheelUpPane send-keys -X clear-selection\; send-keys -X -N 3 scroll-up
bind-key -T copy-mode-vi WheelDownPane send-keys -X clear-selection\; send-keys -X -N 3 scroll-down
```

But keeping the selection would be nice if possible.

**Comment by nicm** [comment]
It would be fine if scroll-up/scroll-down commands did not move the selection.

**Comment by mlgh** [comment]
Hi!
Try launching tmux with this patch: #652 
and add execute something like this
`bind-key -T copy-mode-vi b send -X stop-dragging-selection`
Then in copy mode you can try pressing b and selection will live indepndently from cursor. If you want to modify the selection, press "o" ("other-end" command) and you'll be able to modify selection.

**Comment by mlgh** [comment]
@choco @wincent I believe you are interested in this, could you please try the #652 PR and report if this is the desired behavior for you? Also this PR allows you to move cursor independently from selection, do you see any need in such functionality?
Thanks!

**Comment by choco** [comment]
I was trying this new feature and it works perfectly for a selection created by dragging, but I can't make it work when the selection was created in other ways.
Basically I'm trying to get this to work
```
bind -T copy-mode-vi DoubleClick1Pane select-pane\; send -X select-word\; send -X stop-selection
```
but in this case stop-selection doesn't really stop it, since scrolling after the double click still modifies the selection.

**Comment by mlgh** [comment]
Confirmed. Will try to make a fix now.

**Comment by mlgh** [comment]
@choco could you pleas try with #672 ? Is the behavior expected now?

**Comment by choco** [comment]
@mlgh yep seem to work with that change :)

**Comment by antcodd** [comment]
Hi @mlgh, did a fix for this for select-word/select-line ever get merged? I noticed your PR #672 was declined and ran into this issue recently in 2.5 while trying to make tmux behave like normal applications in terms of selection behaviour.
